spec.html

```diff
@@ -42923,11 +42924,14 @@ <h1>%AsyncFromSyncIteratorPrototype%.throw ( [ _value_ ] )</h1>
             1. Let _O_ be the *this* value.
             1. Assert: _O_ is an Object that has a [[SyncIteratorRecord]] internal slot.
             1. Let _promiseCapability_ be ! NewPromiseCapability(%Promise%).
-            1. Let _syncIterator_ be _O_.[[SyncIteratorRecord]].[[Iterator]].
+            1. Let _syncIteratorRecord_ be _O_.[[SyncIteratorRecord]].
+            1. Let _syncIterator_ be _syncIteratorRecord_.[[Iterator]].
             1. Let _throw_ be GetMethod(_syncIterator_, *"throw"*).
             1. IfAbruptRejectPromise(_throw_, _promiseCapability_).
             1. If _throw_ is *undefined*, then
-              1. Perform ! Call(_promiseCapability_.[[Reject]], *undefined*, &laquo; _value_ &raquo;).
+              1. NOTE: If _syncIterator_ does not have a `throw` method, close it to give it a chance to clean up before we reject the capability.
+              1. Perform ? IteratorClose(_syncIteratorRecord_, ThrowCompletion(_value_)).
```

Comment:
`IteratorClose` step 5 means that you will never get past this line, as written. You probably want to omit `?` here, and just swallow any exceptions from calling `return`.

(Or if you don't want to swallow those exceptions, you need to figure out some other semantics. But IteratorClose does swallow exceptions from `return` when the second argument when is a ThrowCompletion, so that's probably reasonable to match.)

Reply:
Oh yes I definitely want to swallow here, as `IteratorClose` will result in a throw completion (the one we provided as second arg).

Reply:
I think this warrants a NOTE, since it's very unusual to swallow exceptions like this. Just like

```suggestion
              1. Perform IteratorClose(_syncIteratorRecord_, ThrowCompletion(_value_)).
              1. NOTE: any exceptions from the previous line are ignored.
```

Reply:
After double checking, swallowing was wrong, and the way I had setup the iterator close wasn't matching the `yield *` semantics. Updated with forwarding error in case of abrupt close.